staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -221,6 +221,9 @@ func (p *Parser) parseRecursive(cur *ListNode) error {
 	cur.append(newRecursive())
 	if r := p.peek(); isAlphaNumeric(r) {
 		return p.parseField(cur)
+	} else if r == '.' {
+		// one or more periods following the recursive descent operator .. is invalid
```

Comment:
why is one period invalid?

both of these appear equivalent today:
```
kubectl get ns -o 'jsonpath={..spec}'
kubectl get ns -o 'jsonpath={...spec}'
```

that also matches the behavior of https://jsonpath.com/ and both implementations at https://jsonpath.curiousconcept.com/

preventing double-recursion (`....`) seems more reasonable to me

Reply:
I suppose one period following `..` could be considered valid, even if it has no effect. I have an alternate version of the code which ignores any periods immediately following `..`
I think one could argue that is a valid interpretation of what a jsonpath author intended if their expression includes more than 2 periods.

```diff
 // parseRecursive scans the recursive desent operator ..

 func (p *Parser) parseRecursive(cur *ListNode) error {
        p.pos += len("..")
        p.consumeText()
        cur.append(newRecursive())
+       // swallow any periods immediately following the recursive descent operator
+       for r := p.peek(); r == '.'; r = p.peek() {
+               p.pos += len(".")
+               p.consumeText()
+       }
        if r := p.peek(); isAlphaNumeric(r) {
                return p.parseField(cur)
        }
        return p.parseInsideAction(cur)
 }

```

I'm happy to post that version instead if it seems like more appropriate behavior.

Reply:
it is specifically the repeated recursive descent that seems invalid to me

would something like this work:

```diff
diff --git a/staging/src/k8s.io/client-go/util/jsonpath/parser.go b/staging/src/k8s.io/client-go/util/jsonpath/parser.go
index e1aab6804f8..62a2c62abe4 100644
--- a/staging/src/k8s.io/client-go/util/jsonpath/parser.go
+++ b/staging/src/k8s.io/client-go/util/jsonpath/parser.go
@@ -216,6 +216,9 @@ func (p *Parser) parseIdentifier(cur *ListNode) error {
 
 // parseRecursive scans the recursive desent operator ..
 func (p *Parser) parseRecursive(cur *ListNode) error {
+	if lastIndex := len(cur.Nodes) - 1; lastIndex >= 0 && cur[lastIndex].Type() == NodeRecursive {
+		return fmt.Errorf("invalid multiple recursive descent")
+	}
 	p.pos += len("..")
 	p.consumeText()
 	cur.append(newRecursive())
```

Reply:
Yes, that worked (with a slight modification). I have updated the unit tests accordingly.